### PR TITLE
additional buffer argument

### PR DESCRIPTION
**Pull request by ahorek**
https://github.com/jruby/jruby/issues/5242

### CODE REVIEW

File: src/test/ruby/test_cipher.rb [hunk 1]
```diff
@@ -475,4 +475,19 @@ def test_encrypt_aes_cfb_20_incompatibility
...
   end
 
+  def test_encrypt_aes_256_cbc_modifies_buffer
+    cipher = OpenSSL::Cipher.new("AES-256-CBC")
+    cipher.key = "a" * 32
+    cipher.encrypt
+    buffer = ''
+    actual = cipher.update('bar' * 10, buffer)
+    if jruby?
+      expected = "\xE6\xD3Y\fc\xEE\xBA\xB2*\x0Fr\xD1\xC2b\x03\xD0"
```
Comment (ahorek) [inline src/test/ruby/test_cipher.rb]: maybe a bug?

File: src/main/java/org/jruby/ext/openssl/Cipher.java [hunk 2]
```diff
@@ -1104,6 +1104,11 @@ private String getCipherAlgorithm() {
 
     @JRubyMethod
     public IRubyObject update(final ThreadContext context, final IRubyObject arg) {
+        return update(context, arg, null);
+    }
+
+    @JRubyMethod
+    public IRubyObject update(final ThreadContext context, final IRubyObject arg, IRubyObject buffer) {
```
Comment (enebo) [inline src/main/java/org/jruby/ext/openssl/Cipher.java]: As a @JRubyMethod buffer can be any Ruby type.  So you need to do something like TypeConverter.checkStringType.

Comment (ahorek) [reply src/main/java/org/jruby/ext/openssl/Cipher.java]: thanks

File: src/test/ruby/test_cipher.rb (hunk 1 above)
Comment (enebo) [reply src/test/ruby/test_cipher.rb]: @ahorek I just saw this.  I don't know.  I would expect the two impls to generate the same data.

**Comment by janko-m** [comment]
Thanks for the fix! In which JRuby version is this going to be released?

**Pull request merged**